AssemblerLib/MeshComponentMap.cpp

```diff
@@ -259,4 +263,34 @@ std::vector<GlobalIndexType> MeshComponentMap::getGlobalIndicesByComponent(
     return global_indices;
 }
 
+std::vector<GlobalIndexType>
+MeshComponentMap::getIndicesForComponent(
+        const std::vector<GlobalIndexType>& cnt,
+        const unsigned component_id) const
+{
+    std::vector<GlobalIndexType> new_cnt;
+    auto const N = cnt.size() / _num_components;
```

Comment:
In some case, DOF per node can vary. For example, using linear finite element for one PDE (e.g. that of fluid flow) and high order finite element for another PDE (e.g. that of solid deformation) in one geometry element is needed for solving of some coupled equations.